hash_test.go

```diff
@@ -209,12 +275,35 @@ func runMerkleHashTests(t *testing.T, merkleHash func([][]byte) ([]byte, error))
 	}
 }
 
+func runSigningRootTests(t *testing.T, signingRoot func(val interface{}) ([32]byte, error)) {
+	for i, test := range signingRootTests {
+		output1, err := signingRoot(test.val1)
+		if err != nil {
+			t.Fatalf("could not get the signing root of test %d, value 1 %v", i, err)
+		}
+		output2, err := signingRoot(test.val2)
+		if err != nil {
+			t.Fatalf("could not get the signing root of test %d, value 2 %v", i, err)
+		}
+		// Check values have same result hash
+		if !bytes.Equal(output1[:], output2[:]) {
+			t.Errorf("test %d: hash mismatch: %X\n != %X", i, output1, output2)
+		}
+	}
+}
+
 func TestHash(t *testing.T) {
 	runHashTests(t, func(val interface{}) ([32]byte, error) {
 		return TreeHash(val)
 	})
 }
 
+func TestSigningRoot(t *testing.T) {
+	runSigningRootTests(t, func(val interface{}) ([32]byte, error) {
```

Comment:
Please don't use this format, it is a bad pattern that we need to remove.

Reply:
Can we maybe make a tracking issue to fix the tests in this repo?
What would the preferred format be?

Reply:
just run the tests in this function instead of calling another function to run them